domain/ide/workspace/folder.go

```diff
@@ -301,15 +305,20 @@ func (f *Folder) scan(ctx context.Context, path string) {
 	f.scanner.Scan(ctx, path, f.processResults, f.path)
 }
 
-func (f *Folder) processResults(scanData snyk.ScanData) {
+func (f *Folder) processResults(scanData snyk.ScanData, triggerSendAnalytics bool, updateGlobalCache bool) {
 	if scanData.Err != nil {
 		f.sendScanError(scanData.Product, scanData.Err)
 		return
 	}
-	// this also updates the severity counts in scan data, therefore we pass a pointer
-	f.updateGlobalCacheAndSeverityCounts(&scanData)
 
-	go sendAnalytics(&scanData)
+	if updateGlobalCache {
+		// this also updates the severity counts in scan data, therefore we pass a pointer
+		f.updateGlobalCacheAndSeverityCounts(&scanData)
+	}
+
+	if triggerSendAnalytics {
+		go sendAnalytics(&scanData)
```

Comment:
If triggerSendAnalytics were part of `scanData`, `sendAnalytics` could make a local decision, instead of deciding outside of it, whether analytics should be sent. WDYT?